File: src/main/java/org/azamov/learnjakarta/task2/java/session/SessionListener.java
package org.azamov.learnjakarta.task2.java.session;

import javax.servlet.annotation.WebListener;
import javax.servlet.http.HttpSessionEvent;
import javax.servlet.http.HttpSessionListener;

@WebListener
public class SessionListener implements HttpSessionListener {
    @Override
    public void sessionCreated(HttpSessionEvent se) {
        String id = se.getSession().getId();
        System.out.println("created. session id = " + id);
    }

    @Override
    public void sessionDestroyed(HttpSessionEvent se) {
    }
}

File: src/main/java/org/azamov/learnjakarta/task2/kt/session/SessionListener.kt
package org.azamov.learnjakarta.task2.kt.session

import javax.servlet.annotation.WebListener
import javax.servlet.http.HttpSession
import javax.servlet.http.HttpSessionEvent
import javax.servlet.http.HttpSessionListener

@WebListener
class SessionListener:HttpSessionListener {
    override fun sessionCreated(se: HttpSessionEvent?) {
        super.sessionCreated(se)
        val session =se?.session?:return
        val id =session.id
        println("created. session id = $id")
    }

    override fun sessionDestroyed(se: HttpSessionEvent?) {
        super.sessionDestroyed(se)
    }
}